tools/clang/lib/SPIRV/SpirvContext.cpp

```diff
@@ -527,5 +527,23 @@ void SpirvContext::moveDebugTypesToModule(SpirvModule *module) {
   typeTemplateParams.clear();
 }
 
+const SpirvIntrinsicType *SpirvContext::getSpirvIntrinsicType(
+    unsigned typeId, unsigned typeOpCode,
+    llvm::ArrayRef<SpvIntrinsicTypeOperand> operands) {
+  if (spirvIntrinsicTypes[typeId] == nullptr) {
+    spirvIntrinsicTypes[typeId] =
+        new (this) SpirvIntrinsicType(typeOpCode, operands);
```

Comment:
We have to free this allocation in the destructor of `SpirvContext`.
In `SpirvContext::~SpirvContext()`, please add
```
for (auto itr: spirvIntrinsicTypes)
  if (itr->second)
   delete itr->second;
```